exodus/analysis_query/models.py

```diff
@@ -46,11 +48,26 @@ def validate_handle(handle):
             _('Too much pending requests, please retry later'))
 
 
+def validate_file_extension(value):
+    import os
+    from django.core.exceptions import ValidationError
+    ext = os.path.splitext(value.name)[1]  # [0] returns path+filename
+    valid_extensions = ['.apk', ]
+    if not ext.lower() in valid_extensions:
+        raise ValidationError(_('Unsupported file extension.'))
+
+
 class AnalysisRequest(models.Model):
     uploaded_at = models.DateTimeField(auto_now_add=True)
     bucket = models.CharField(max_length=200, default='')
-    handle = models.CharField(max_length=500, validators=[validate_handle])
+    handle = models.CharField(max_length=500, default='', blank=True, validators=[validate_handle])
```

Comment:
Out of curiosity, what does `blank=True` do here?

Reply:
`blank=True` allows empty value for the handle. In the case of application upload, there is no handle specified by the user.